hr_holidays_remaining_leaves/models/hr_leave.py

```diff
@@ -46,9 +45,6 @@ def _compute_remaining_leaves(self):
                     ("employee_id", "=", allocation.employee_id.id),
                     ("state", "=", "validate"),
                     ("holiday_status_id", "=", allocation.holiday_status_id.id),
-                    "|",
-                    ("holiday_allocation_id", "=", allocation.id),
-                    ("holiday_allocation_id", "=", False),
```

Comment:
you remove these 2 domains because the field `holiday_allocation_id` was removed from hr.leave.
But, this causes a new issue:
- I create 2 records of hr.leave.allocation:
  + A1: From 2024-05-01 to unlimited, duration = 10 days.
  + A2: From 2024-06-01 to unlimited, duration = 5 days.
- Create a new leave from 2024-06-10 to 2024-06-12. Both A1 and A2 reduces their remaining leave.
=> This's incorrect.

Reply:
Branch updated, in this fixed I using _get_consumed_leaves(), this method return remain leaves and take current date as params for calculated( params `ignored_future`)